packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -1317,6 +1326,17 @@ abstract class State<T extends StatefulWidget> with Diagnosticable {
   @mustCallSuper
   void deactivate() { }
 
+  /// Called when this object is reactivate.
+  ///
+  /// If [widget] or ancestor widget has GlobalKey possible to call [activate]
+  /// after [deactivate] called.
+  ///
+  /// If you override this, make sure to end your method with a call to
+  /// super.activate().
+  @protected
+  @mustCallSuper
+  void activate() { }
```

Comment:
i think having this is reasonable.

Could we rename it "reactivate", to make it clear that it's not called when the State is first created?

Reply:
I was thinking the same thing, but should `Element.activate` be rename as well?